### ISSUE TITLE
When importing `@stylistic/eslint-plugin-ts` the files pattern is ignored.

### ISSUE THREAD

**Issue by srh-cuidate** [eslint-stylistic/eslint-stylistic#771]
Hey there, I am having this weird issue when dynamically importing the `@stylistic/eslint-plugin-ts`.

First, I am dynamically importing the module because of a `[ERR_REQUIRE_ESM]: require() of ES Module` problem:
```js
const globals = require('globals');
const { FlatCompat } = require('@eslint/eslintrc');
const importPlugin = require('eslint-plugin-import');
const prettierPlugin = require('eslint-plugin-prettier');
const prettierConfig = require('eslint-plugin-prettier/recommended');
const eslint = require('@eslint/js');
// Only '@stylistic/eslint-plugin-ts' fails.
const stylisticTsPlugin = require('@stylistic/eslint-plugin-ts');
const tseslint = require('typescript-eslint');
```

Then when I try to import like this:
```js
(async () => {
  const { default: stylisticTsPlugin } = await import('@stylistic/eslint-plugin-ts');

  module.exports = tseslint.config(...);
})();
```
I'm getting this issue:
```
Error importing module: Cannot read properties of undefined (reading 'all')

Oops! Something went wrong! :(

ESLint: 9.18.0

You are linting "{.setup,src,test}/**/*.ts", but all of the files matching the glob pattern "{.setup,src,test}/**/*.ts" are ignored.

If you don't want to lint these files, remove the pattern "{.setup,src,test}/**/*.ts" from the list of arguments passed to ESLint.

If you do want to lint these files, explicitly list one or more of the files from this glob that you'd like to lint to see more details about why they are ignored.

  * If the file is ignored because of a matching ignore pattern, check global ignores in your config file.
    https://eslint.org/docs/latest/use/configure/ignore

  * If the file is ignored because no matching configuration was supplied, check file patterns in your config file.
    https://eslint.org/docs/latest/use/configure/configuration-files#specifying-files-with-arbitrary-extensions

  * If the file is ignored because it is located outside of the base path, change the location of your config file to be in a parent directory.
```

My whole `eslint.config.js` is:
```js
const globals = require('globals');
const { FlatCompat } = require('@eslint/eslintrc');
const importPlugin = require('eslint-plugin-import');
const prettierPlugin = require('eslint-plugin-prettier');
const prettierConfig = require('eslint-plugin-prettier/recommended');
const eslint = require('@eslint/js');
// const stylisticTsPlugin = require('@stylistic/eslint-plugin-ts');
const tseslint = require('typescript-eslint');
const prettierParams = require('./prettier.config');

const compat = new FlatCompat({
  baseDirectory: __dirname,
});

(async () => {
  try {
    // FIXME: Find out why when importing this module the files pattern is ignored.
    const { default: stylisticTsPlugin } = await import('@stylistic/eslint-plugin-ts');

    module.exports = tseslint.config(
      eslint.configs.recommended,
      ...tseslint.configs.recommended,
      ...compat.extends('eslint-config-semistandard'),
      ...stylisticTsPlugin.configs['all'],
      prettierConfig,
      {
        files: ['{.setup,src,test}/**/*.{js,ts}'],
        languageOptions: {
          ecmaVersion: 2020,
          globals: globals.node,
        },
        linterOptions: {
          noInlineConfig: false,
          reportUnusedDisableDirectives: true,
        },
        plugins: {
          '@stylistic/ts': stylisticTsPlugin,
          import: importPlugin,
          prettier: prettierPlugin,
        },
        rules: {
          // '@stylistic/ts/array-element-newline': 'off',
          // '@stylistic/ts/arrow-parens': 'off',
          // '@stylistic/ts/comma-dangle': 'off',
          // '@stylistic/ts/dot-location': 'off',
          // '@stylistic/ts/function-call-argument-newline': 'off',
          // '@stylistic/ts/function-paren-newline': 'off',
          // '@stylictic/js/implicit-arrow-linebreak': 'off',
          // '@stylistic/ts/indent': 'off',
          // '@stylistic/ts/multiline-ternary': 'off',
          // '@stylistic/ts/newline-per-chained-call': 'off',
          // '@stylistic/ts/no-extra-parens': 'off',
          // '@stylistic/ts/object-curly-spacing': 'off',
          // '@stylistic/ts/object-property-newline': 'off',
          // '@stylistic/ts/padded-blocks': 'off',
          // '@stylistic/ts/quote-props': 'off',
          // '@stylistic/ts/quotes': ['error', 'single'],
          // '@stylistic/ts/space-before-function-paren': 'off',
          // '@stylistic/ts/wrap-iife': 'off',
          // '@stylistic/ts/wrap-regex': 'off',
          '@typescript-eslint/no-unused-vars': 'off',
          'no-throw-literal': 'off',
          'no-useless-constructor': 'off',
          'no-useless-escape': 'off',
          'prettier/prettier': [
            'error',
            {
              ...prettierParams,
              endOfLine: 'auto',
              semi: true,
            },
            { usePrettierrc: false },
          ],
        },
      },
      {
        files: ['lib/**/*.entity.ts'],
        rules: { '@stylistic/js/indent': 'off' },
      },
    );
  } catch (error) {
    console.error("Error importing module:", error.message);
  }
})();

```

**Comment by keijokapp** [comment]
> First, I am dynamically importing the module because of a [ERR_REQUIRE_ESM]: require() of ES Module problem:

This is not going to work. ESLint configuration can't be specified asynchronously with CommonJS.

Do not try to hack around using ESM. Just migrate to ESM. You don't have to migrate the whole project - just the ESLint configuration file. It's trivial in most cases - convert `requires`-s to imports, `module.exports` to `export default` and rename the configuration file to `eslint.config.mjs`.

**Comment by srh-cuidate** [comment]
> > First, I am dynamically importing the module because of a [ERR_REQUIRE_ESM]: require() of ES Module problem:
> 
> This is not going to work. ESLint configuration can't be specified asynchronously with CommonJS.
> 
> Do not try to hack around using ESM. Just migrate to ESM. You don't have to migrate the whole project - just the ESLint configuration file. It's trivial in most cases - convert `requires`-s to imports, `module.exports` to `export default` and rename the configuration file to `eslint.config.mjs`.

Many thanks! It worked
